overlord/snapstate/handlers.go

```diff
@@ -2419,14 +2427,28 @@ func (m *SnapManager) doDiscardSnap(t *state.Task, _ *tomb.Tomb) error {
 	// drop any potential revert status for this revision
```

Comment:
I would prefer an approach where we change doLinkSnap and undoLinkSnap instead of doDiscardSnap, we can discuss this

Reply:
Thanks for the feedback. It now stores the revisions before the old candidate index and compares those with the Sequence in undo, so all the logic is in (un)doLinkSnap